spec.md

```diff
@@ -703,6 +704,36 @@ Probe types may target `const`{.firrtl} signals, but cannot use
 `RWProbe<const T>`{.firrtl}, as constant values should never be mutated at
 runtime.
 
+## Type Alias
+
+A type alias provides a way to assign names to existing FIRRTL types, enabling
+their reuse across multiple declarations and potentially serving as hints in
+verilog code generation.
```

Comment:
Nits: use stronger, direct language here. "A type alias is", "Type aliases enable", etc.

The "potentially serving as hints may be confusing. (Is this mandatory, is this optional, etc.?) You may want to leave this out and adds this to the ABI doc when that lands.